WhateverGreen/kern_igfx_backlight.cpp

```diff
@@ -239,26 +240,39 @@ bool BrightnessRequestEventSource::checkForWork() {
 	uint32_t current = IGFX::callbackIGFX->readRegister32(request.controller, request.address);
 	uint32_t cbrightness = request.getCurrentBrightness(current);
 	uint32_t tbrightness = request.getTargetBrightness();
+
+	if (cbrightness == tbrightness) {
+		DBGLOG("igfx", "BLS: [COMM] The request is already completed. Will wait for the next invocation.");
+		return false;		
+	}
+	
 	tbrightness = max(tbrightness, smoother->brightnessRange.first);  // Ensure that target >= lowerbound
 	tbrightness = min(tbrightness, smoother->brightnessRange.second); // Ensure that target <= upperbound
 	uint32_t distance = max(cbrightness, tbrightness) - min(cbrightness, tbrightness);
-	uint32_t stride = distance / smoother->steps + ((distance % smoother->steps) ? 1 : 0);
-	DBGLOG("igfx", "BLS: [COMM] Processing the request: Current = 0x%08x; Target = 0x%08x; Distance = %04u; Steps = %u; Stride = %u.",
-		   cbrightness, tbrightness, distance, smoother->steps, stride);
+	DBGLOG("igfx", "BLS: [COMM] Processing the request: Current = 0x%08x; Target = 0x%08x; Distance = %04u; Steps = %u.",
+		   cbrightness, tbrightness, distance, smoother->steps);
 	
 	// Process the request
 	if (distance > smoother->threshold) {
 		if (cbrightness < tbrightness) {
 			// Increase the brightness
-			for (uint32_t value = cbrightness + stride; value <= tbrightness - stride; value += stride) {
+			for (uint32_t i = 1; i < smoother->steps; i++) {
+				uint32_t value = cbrightness + smoother->sines[i] * distance;
 				IGFX::callbackIGFX->writeRegister32(request.controller, request.address, request.getTargetRegisterValue(value));
 				IOSleep(smoother->interval);
+				if (smoother->request.id != request.id) {
+					return true; // Start again, because there is a new request
+				}
 			}
 		} else if (cbrightness > tbrightness) {
 			// Decrease the brightness
-			for (uint32_t value = cbrightness - stride; value >= tbrightness + stride; value -= stride) {
+			for (uint32_t i = 1; i < smoother->steps; i++) {
```

Comment:
Leave a comment that 1 is intended here.

Reply:
Done